astropy/units/tests/test_units.py

```diff
@@ -58,7 +58,17 @@ def test_invalid_compare():
 
 
 def test_convert():
-    assert u.h._get_converter(u.s)(1) == 3600
+    assert u.h.get_converter(u.s)(1) == 3600
+
+
+def test_convert_roundtrip():
+    c1 = u.cm.get_converter(u.m)
+    c2 = u.m.get_converter(u.cm)
+    assert c1(c2(10.0)) == c2(c1(10.0))
```

Comment:
Both these are risky with rounding errors; maybe use `np.isclose` (can set `rtol=1e-15` and `atol=0`
(though the tests pass and your call is nicely symmetric, so maybe fine to keep as is)